provider/internal/service/duoadminpanel/resource.go

```diff
@@ -0,0 +1,189 @@
+package duoadminpanel
+
+import (
+	"context"
+	"fmt"
+	"github.com/CiscoDevnet/terraform-provider-cdo/internal/util"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/setdefault"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	cdoClient "github.com/CiscoDevnet/terraform-provider-cdo/go-client"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+var _ resource.Resource = &Resource{}
+
+func NewResource() resource.Resource {
+	return &Resource{}
+}
+
+type Resource struct {
+	client *cdoClient.Client
+}
+
+type ResourceModel struct {
+	Id             types.String `tfsdk:"id"`
+	Name           types.String `tfsdk:"name"`
+	IntegrationKey types.String `tfsdk:"integration_key"`
+	SecretKey      types.String `tfsdk:"secret_key"`
+	Host           types.String `tfsdk:"host"`
+	Labels         types.Set    `tfsdk:"labels"`
+}
+
+func (r *Resource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_duo_admin_panel"
+}
+
+func (r *Resource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "Example resource description.",
+
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				MarkdownDescription: "Unique identifier of the Duo Admin Panel. This is a UUID and is automatically generated when the Duo Admin Panel is created.",
+				Computed:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"name": schema.StringAttribute{
+				MarkdownDescription: "A human-readable name for the Duo Admin Panel.",
+				Required:            true,
+			},
+			"integration_key": schema.StringAttribute{
+				MarkdownDescription: "The integration key of Duo Admin Panel.",
+				Required:            true,
+				Sensitive:           true,
+			},
+			"secret_key": schema.StringAttribute{
+				MarkdownDescription: "The secret key of the Duo Admin Panel.",
+				Required:            true,
+				Sensitive:           true,
+			},
+			"host": schema.StringAttribute{
```

Comment:
Please rename to `api_hostname` to correspond to what the user will see in the Duo UI

Reply:
As discussed offline, we decided to go with `host` to stay consistent with other resources because it is relatively easy for user to figure out `api_hostname == host`